rotkehlchen/chain/ethereum/modules/eth2/decoder.py

```diff
@@ -65,23 +66,27 @@ def _decode_eth2_deposit_event(self, context: DecoderContext) -> DecodingOutput:
                 event.event_type == HistoryEventType.SPEND and
                 event.event_subtype == HistoryEventSubType.NONE and
                 event.asset == A_ETH and
-                event.balance.amount == amount
+                event.balance.amount >= amount
```

Comment:
for this contract it is the only method you can call. In this case I believe you can use

```
event.balance.amount == amount * len(context.all_logs)
```

have you checked it?

Reply:
I would not do this due to 2 reasons:
1. It would limit us to a particular contract method. Others may have different events emitted.
2. Doing the check that you suggested will be possible only for the first deposit event. Since after each deposit event processed `event.balance.amount` is reduced, the amount changes. And trying to guess the right amount I think is not a good idea since it would also require to rely on the structure of logs emitted by contracts, as mentioned in (1).

Reply:
I would not do this due to 2 reasons:
1. It would limit us to a particular contract method. Others may have different events emitted.
2. Doing the check that you suggested will be possible only for the first deposit event. Since after each deposit event processed `event.balance.amount` is reduced, the amount changes. And trying to guess the right amount I think is not a good idea since it would also require to rely on the structure of logs emitted by contracts, as mentioned in (1).

Reply:
later I saw the play you make with the amounts. For 1. you are already limitating it since is a rule for one address. Only this contract will process this logic and this contract has only this method